cwl/src/test/scala/wdl4s/cwl/CwlWorkflowWomSpec.scala

```diff
@@ -0,0 +1,301 @@
+package wdl4s.cwl
+
+import cats.data.Validated.{Invalid, Valid}
+import org.scalatest.{FlatSpec, Matchers}
+import wdl4s.wdl.types.{WdlFileType, WdlStringType}
+import wdl4s.wom.callable.Callable.RequiredInputDefinition
+import wdl4s.wom.callable.{TaskDefinition, WorkflowDefinition}
+import wdl4s.wom.executable.Executable
+import wdl4s.wom.graph.{CallNode, GraphInputNode, GraphOutputNode, RequiredGraphInputNode}
+
+class CwlWorkflowWomSpec extends FlatSpec with Matchers {
+  "munging the workflow output id" should "remove the filename" in {
+    val id = "file:///home/dan/common-workflow-language/v1.0/examples/1st-workflow.cwl#untar/example_out"
+
+    val out = WorkflowOutputsToOutputDefinition.mungeId(id)
+
+    out shouldBe "example_out"
+  }
+
+  "munging the runnable id" should "remove the filename" in {
+    val id = "file:///home/dan/common-workflow-language/v1.0/examples/tar-param.cwl#example_out"
+
+    val out = RunOutputsToTypeMap.mungeId(id)
+
+    out shouldBe "example_out"
+  }
+
+  "munging runnable output id " should "be able to skip the path args" in {
+    val id = "file:///home/dan/common-workflow-language/v1.0/examples/tar-param.cwl#ps/0b4ba500-5584-4fed-a831-9fa6f914ad3f/ps-stdOut"
+
+    val out = RunOutputsToTypeMap.mungeId(id)
+
+    out shouldBe "ps-stdOut"
+  }
+
+
+  "A Cwl object for 1st-tool" should "convert to WOM" in {
+    val firstTool =
+      """
+cwlVersion: v1.0
+class: CommandLineTool
+baseCommand: echo
+inputs:
+- type: string
+  inputBinding:
+    position: 1
+  id: message
+outputs: []
+""".stripMargin
+
+    CwlCodecs.decodeCwl(firstTool) match {
+      case Right((clt: CommandLineTool, _)) =>
+        clt.womExecutable match {
+          case Valid(wom) =>
+            wom.entryPoint match {
+              case workflow: TaskDefinition =>
```

Comment:
Is this three-`match` logic going to be needed every time anyone runs `decodeCwl`? 

My friend, Naïve Chris would say "Surely it's either valid or it isn't?"